### DISCUSSION TITLE
Rgba is not working in text rendering in debian 11

### DISCUSSION THREAD

**Discussion by Ruchika-K** [libvips/libvips#3003] [Q&A]
So I have been using govips which uses libvips internally for development of a service.
I am rendering text in C using cgo :-

`vips_text(out,text,"font",font,"dpi",dpi,"width",width,"rgba",TRUE,NULL)`

This works perfectly fine on Mac.

But when this is compiled for debian 11(bullseye) - with the following dependencies in the machine :
1. Golang 1.19
2. Pkg-config latest version
3. libvips-dev latest version
4. gcc latest version

And the executable is run , it fails with no error but doesn't create the vips image, i.e. out is nil

I noticed that if rgba param is removed, it works. 

Any idea why this would be happening? Is there any dependency missing ?

I checked that pangocairo is present in pkg-config

Note that setting rgba to FALSE also fails.

**Comment by jcupitt** [comment]
Hi again,

I made a dockerfile for debian11:

https://github.com/jcupitt/docker-builds/tree/master/libvips-debian11

It seems to work for me:

```
$ docker run --rm -it -v $PWD:/data libvips-debian11
root@7d1284a3c986:/usr/local/src# vips text x.png hello --rgba
root@7d1284a3c986:/usr/local/src# vipsheader x.png
x.png: 27x10 uchar, 4 bands, srgb, pngload
root@7d1284a3c986:/usr/local/src# 
exit
```

To make:

![x](https://user-images.githubusercontent.com/580843/185778353-4cb13d13-3a78-441b-9522-4e4876298093.png)

**Reply by jcupitt** [reply]
Oh, hang on, are you using the debian libvips package? Debian 11 includes libvips 8.10 from a few years ago. You need 8.11 or later for the "rgba" option.

**Reply by Ruchika-K** [reply]
Hi ,
Yes I am using libvips 8.10.5 in Deb 11.
Is libvips 8.11 available? Couldn't find it.
Is there any other way to let rgba work?
Installing some other package which is required separately?
I had tried installing  libpangocairo-1.0.0 as well . But that didn't work.

**Reply by Ruchika-K** [reply]
I also tried using the same docker file which you have used to create the executable. It still didn't work.

**Reply by jcupitt** [reply]
Debian stable is always a few years behind development. You'll need to build libvips yourself, or switch to another platform. You shouldn't really use the debian package anyway -- it includes a lot of unsafe image loaders.

> I also tried using the same docker file which you have used to create the executable. It still didn't work.

You need to say what you did and what errors you saw.

```
$ docker build -t libvips-debian11 .
$ docker run --rm -it -v $PWD:/data libvips-debian11
# cd /data; vips text x.png hello --rgba
```

is working here.

**Reply by Ruchika-K** [reply]
There is no error in the C code. It is just returning nil image. So while using it's ref in govips  it throws null pointer exception.

**Reply by Ruchika-K** [reply]
For building our own package for production purpose what are the required dependencies to include? Will the dependencies used in the docker file mentioned sufficient?

**Reply by jcupitt** [reply]
> There is no error in the C code. It is just returning nil image.

It sounds like you have two libvipes installed and it's not picking up the right one at link time. I would look into the linker paths.

> For building our own package for production purpose what are the required dependencies to include? 

Only glib and expat are required, all others are optional, so it depends what features you need. I would make a table of the file formats you need to support and the features you require. The README has the details.

**Reply by Ruchika-K** [reply]
Okay thanks @jcupitt .
Yes the docker file I was using was only to build. The executable was still running on older version machine.
It should run properly once those dependencies are installed in the machine we need to run it 
I'll build the custom package rather than using libvips-dev directly.
